src/components/Courses/components/CourseCard/CourseCard.jsx

```diff
@@ -8,24 +8,29 @@ import { formatTime } from '../../../../helpers/getSourseDuration';
 import trash from '../../../../assets/trash.png';
 import edit from '../../../../assets/edit.png';
 import { SHOW_COURSE_LABEL } from '../../../../constants';
-import { deleteCourseAction } from '../../../../store/courses/actions';
-import { deleteCourse } from '../../../../services';
+import { deleteCourse } from '../../../../store/courses/thunk';
+import { getUser } from '../../../../store/selectors';
 
 import styles from './CourseCard.module.css';
 
 const CourseCard = (props) => {
 	const dispatch = useDispatch();
 	const navigate = useNavigate();
+	const user = useSelector(getUser);
 
-	const handleDeleteCourse = async (id) => {
-		const deleteResponse = await deleteCourse(id);
-		if (deleteResponse.successful) {
-			dispatch(deleteCourseAction(id));
-		}
+	const handleDeleteCourse = (id) => {
+		dispatch(deleteCourse(id));
+	};
+	const handleUpdateCourse = (id) => {
+		navigate(`/courses/update/${id}`);
+	};
+
+	const handleInfoClick = (id) => {
+		navigate(`/courses/${id}`);
 	};
 
-	const handleInfoClick = (course) => {
-		navigate(`/courses/${course.id}`);
+	const isUserAdmin = () => {
+		return user?.role?.toLowerCase() === 'admin';
```

Comment:
You have that check in several places, This is definitely a good candidate to move to the selector in one place. It will be easy extend and maintain in the feature